src/main/java/org/opensearch/index/codec/customcodecs/Lucene912QatStoredFieldsFormat.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.codec.customcodecs;
+
+import org.apache.lucene.codecs.StoredFieldsFormat;
+import org.apache.lucene.codecs.StoredFieldsReader;
+import org.apache.lucene.codecs.StoredFieldsWriter;
+import org.apache.lucene.codecs.compressing.CompressionMode;
+import org.apache.lucene.codecs.lucene90.compressing.Lucene90CompressingStoredFieldsFormat;
+import org.apache.lucene.index.FieldInfos;
+import org.apache.lucene.index.SegmentInfo;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.store.IOContext;
+
+import java.io.IOException;
+import java.util.Objects;
+import java.util.function.Supplier;
+
+import com.intel.qat.QatZipper;
+
+import static org.opensearch.index.codec.customcodecs.backward_codecs.lucene99.Lucene99QatCodec.DEFAULT_COMPRESSION_LEVEL;
+import static org.opensearch.index.codec.customcodecs.backward_codecs.lucene99.Lucene99QatCodec.DEFAULT_QAT_MODE;
+
+/** Stored field format used by pluggable codec */
+public class Lucene912QatStoredFieldsFormat extends StoredFieldsFormat {
```

Comment:
Ideally we shouldn't create a new stored fields format file if the underlying functionality isn't changing across new codecs. Like Lucene is using an [old stored fields format from Lucene90](https://github.com/apache/lucene/blob/main/lucene/backward-codecs/src/java/org/apache/lucene/backward_codecs/lucene912/Lucene912Codec.java#L121). 

But I believe, since modes are tightly coupled as codec names, it becomes a bit difficult. I'm open to hear your thoughts on this, will think more on it as well.

Reply:
> But I believe, since modes are tightly coupled as codec names, it becomes a bit difficult. I'm open to hear your thoughts on this, will think more on it as well.

Right, the coupling to mode (and to the codec)  makes is difficult to reuse even if the underlying functionality isn't changing